source/_integrations/pi4ioe5v9xxxx.markdown

```diff
@@ -0,0 +1,136 @@
+
+## Binary Sensor
+
+The `pi4ioe5v9xxxx` binary sensor platform allows you to read sensor values from the I/O pins of your I/O expander.
+
+The pin numbers are from 1 to X where: 1-8 correspond to port 0 (00-07) and 9-16 to port 1, etc.
+
+### Configuration
+
+To use the I/O pins of an pi4ioe5v9xxxx connected to an I2C bus of your Raspberry Pi as binary sensors, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+binary_sensor:
+- platform: pi4ioe5v9xxxx
+   i2c_bus: 1
+   pins:
+     01: Pin_01/PI0_0
+     02: Pin_02/PI0_1
+     03: Pin_03/PI0_2
+     04: Pin_04/PI0_3
+     05: Pin_05/PI0_4
+     06: Pin_06/PI0_5
+     07: Pin_07/PI0_6
+     08: Pin_08/PI0_7
+```
+{% configuration %}
+i2c_bus:
+  description: i2c bus containing the pi4ioe5v9xxxx chip.
+  required: true
+  type: integer
+  default: "`1`"
+i2c_address:
+  description: i2c address of pi4ioe5v9xxxx chip.
+  required: false
+  type: integer
+  default: "`0x20`"
+bits:
+  description: number of bits of pi4ioe5v9xxxx chip, see particular datasheet for your device for the right number.
+  required: false
+  type: integer
+  default: "`24`"
+pins:
+  description: List of used pins.
+  required: true
+  type: map
+  keys:
+    "pin: name":
+      description: The pin numbers (from 1 to X) and corresponding names.
+      required: true
+      type: [integer, string]
+invert_logic:
+  description: If `true`, inverts the output logic to ACTIVE LOW.
+  required: false
+  type: boolean
+  default: "`false` (ACTIVE HIGH)"
+{% endconfiguration %}
+
+
+## Switch
+
+The `pi4ioe5v9xxxx` switch platform allows you to write to the I/O pins of your I2C I/O expander.
+
+The pin numbers are from 1 to X where: 1-8 correspond to port A (A1-A8) and 9-16 to port B (B1-B8), etc.
+
+### Configuration
+
+To use the I/O pins of an pi4ioe5v9xxxx connected to an I2C bus as switches, add the following to your `configuration.yaml` file:
+
+```yaml
+# Example configuration.yaml entry
+switch:
+  - platform: pi4ioe5v9xxxx
+    i2c_bus: 1
+    pins:
+     01: Pin_01/PI0_0
+     02: Pin_02/PI0_1
+     03: Pin_03/PI0_2
+     04: Pin_04/PI0_3
+     05: Pin_05/PI0_4
+     06: Pin_06/PI0_5
+     07: Pin_07/PI0_6
+     08: Pin_08/PI0_7
+```
+
+{% configuration %}
+i2c_bus:
+  description: i2c bus containing the pi4ioe5v9xxxx chip.
+  required: true
+  type: integer
+  default: "`1`"
+i2c_address:
+  description: i2c address of pi4ioe5v9xxxx chip.
+  required: false
+  type: integer
+  default: "`0x20`"
+bits:
+  description: number of bits of pi4ioe5v9xxxx chip, see particular datasheet for your device for the right number.
+  required: false
+  type: integer
+  default: "`24`"
+pins:
+  description: List of used pins.
+  required: true
+  type: map
+  keys:
+    "pin: name":
+      description: The pin numbers (from 1 to X) and corresponding names.
+      required: true
+      type: [integer, string]
```

Comment:
Would the following be clearer than [integer, string]  (both here in and in the binary_sensor)?

      type: [integer] : [string]

Reply:
@mrand That suggested type is invalid.